### PR TITLE
Rework Stop Dropdown (Again)

### LINKED ISSUE #407: Rework Stop Dropdown
#384 needs a few more tweaks.

### PR DESCRIPTION
**Pull request by akaplo**
Closes #407 in a continuation of #384.

**Left:** Android styling for sort-by-time **Right**: iOS styling for sort-by-`RouteDirection` with 1 Route expanded
image

**The desktop version**
image

### CODE REVIEW

**Comment by mboneil10** [comment]
I merged with master. Was there a code-related reason you hadn't done so? I figured it would be okay, but the commit can be reverted if needed.

File: src/pages/stop/stop.html [hunk 1]
```diff
@@ -31,28 +31,33 @@
     <ion-refresher-content pullingText="Pull to refresh departures...">
     </ion-refresher-content>
   </ion-refresher>
-  <ion-list aria-label="upcoming departures at this stop" role="list" no-lines>
+  <ion-list aria-label="upcoming departures at this stop" role="list">
```
Comment (mboneil10) [inline src/pages/stop/stop.html:34]: I agree with this change. Great!

**Comment by mboneil10** [comment]
@werebus has mentioned [Material Designs specs](https://material.io/guidelines/components/lists-controls.html#lists-controls-types-of-menu-controls) before, and I think in this case I see something that could be changed. The stop dropdown arrows could be pointing down when they are collapsed and up when they are expanded. This way, we'd indicate that this item is expandable, instead of a menu item.

File: src/pages/stop/stop.html [hunk 2]
```diff
@@ -83,14 +83,23 @@
               </ion-col>
               <ion-col>
                 <div class="center departure-padding">
-                  {{departure.Times.sExact}}
+                  {{departure.Times.sExact}}<br>
```
Comment (mboneil10) [inline src/pages/stop/stop.html]: No other way to do this than a `br tag`?

File: src/pages/stop/stop.html [hunk 3]
```diff
@@ -61,24 +61,24 @@
           </ion-grid>
         </div>
         <!-- When a route has been expanded, show its departures -->
-        <div ion-item [hidden]="!isRouteDropdownShown(direction)" (click)="goToRoutePage(direction.RouteId)">
+        <div [style.background-color]="'#'+routeList[direction.RouteId]?.Color" ion-item [hidden]="!isRouteDropdownShown(direction)" (click)="goToRoutePage(direction.RouteId)">
```
Comment (mboneil10) [inline src/pages/stop/stop.html]: Have you seen how this looks with all of the route colors?

File: src/pages/stop/stop.html [hunk 4]
```diff
@@ -35,7 +35,6 @@
     <div *ngIf="order == '0'">
       <div *ngFor="let direction of departuresByDirection">
         <div ion-item (click)="toggleRouteDropdown(direction)" [style.background-color]="'#'+routeList[direction.RouteId]?.Color" [style.color]="'white'">
-          <!-- [ngClass]="{active: isRouteDropdownShown({{direction}})}" -->
```
Comment (mboneil10) [inline src/pages/stop/stop.html:38]: It's nice that you did this in a completely separate commit. Thanks!

**Review by mboneil10 — CHANGES_REQUESTED** [review]
I mentioned that the dropdown doesn't look like a dropdown due to the orientation of the arrows. All of the other changes are great.

**Comment by mboneil10** [comment]
Another thing: OOC, when did we start displaying destinations? What is the purpose? For Bus Garage destinations, or...?

**Comment by akaplo** [comment]
Good question - when ordering departures by `Direction`, there can be multiple departures with the same direction but different destinations, so it's nice to specify the destination in each entry.

The first screenshot in this PR shows an example - two 31s are leaving Studio Arts, one to Sunderland and one to the Garage. On the left, we see them both cause we order by time, but on the right they're grouped together, because they're both headed North.

**Comment by dfaulken** [comment]
Bus Garage is the major case for UMass, but on the whole, the other two PVTA properties have more variants per route than we do. So yes, buses can be leaving from the same stop, on the same route, and in the same direction — but still be going different places once they depart.

**Comment by mboneil10** [comment]
Alright, sounds good. Thank you. @dfaulken @akaplo

**Comment by mboneil10** [comment]
Do you have an updated screenshot you can share @akaplo?

**Comment by akaplo** [comment]
@mboneil10 good point about the arrow directions.

**Comment by akaplo** [comment]
Arrows and screenshots updated. @mboneil10 or anyone else - any final thoughts?

**Review by mboneil10 — APPROVED** [review]

**Comment by mboneil10** [comment]
Wait wait wait, it's the other way.

**Review by mboneil10 — CHANGES_REQUESTED** [review]
Sorry, the arrow should face up when expanded and down when collapsed

**Comment by mboneil10** [comment]
I can make this fix for you @akaplo

**Comment by mboneil10** [comment]
screen shot 2017-08-30 at 9 59 27 am

**Comment by mboneil10** [comment]
screen shot 2017-08-30 at 10 01 09 am

**Comment by akaplo-cb** [comment]
Woops, my bad! Thanks @mboneil10. 👍

**Comment by akaplo** [comment]
Woops! My bad. Thanks @mboneil10 👍

**Review by mboneil10 — APPROVED** [review]

**Pull request merged**